src/auth/dto/create-user.dto.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  IsString,
+  MinLength,
+  Matches,
+  IsNotEmpty,
+  IsEmail,
+  Length,
+} from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+export class CreateUserDto {
+  @IsString()
+  @ApiProperty({
+    example: 'username',
+  })
+  @IsNotEmpty({ message: 'First name is required' })
+  name: string;
+
+  @Length(8, 18, {
+    message: 'Phone number should be between 8 and 18 digits',
+  })
+  // @Matches(/^\+\d{1,3}-?\d{7,15}$/, {
+  //   message:
+  //     'Phone number should start with a + and country code (1 to 3 digits), followed by an optional dash, and then 7 to 15 digits',
+  // })
+  @ApiProperty({
+    description:
+      'The phone number of the user, including country code and an optional dash',
```

Comment:
PullProofAI

The commented-out code for phone number validation is unnecessary and can be removed.

```diff
-   // @Matches(/^\+\d{1,3}-?\d{7,15}$/, {
-   //   message:
-   //     'Phone number should start with a + and country code (1 to 3 digits), followed by an optional dash, and then 7 to 15 digits',
-   // })
```

#### Issue:

46:
The regex pattern for password validation might be too complex and could be simplified for better readability and maintainability.

#### Suggested fix:

You can simplify the regex pattern for password validation to improve readability:

```typescript
45:   @Matches(/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}$/, {
46:     message: "Password must contain at least one digit, one lowercase letter, one uppercase letter, and be at least 8 characters long",
47:   })
```

These changes will help streamline the code and make it more maintainable.


<!-- This is an auto-generated comment by PullProofAI -->